engine/robot.py

```diff
@@ -330,10 +361,63 @@ def set_phase(self, new_phase):
         with open(CSV_PATH + '/phases.csv', 'a') as fd:
             fd.write(str(self.phase) + '\n')
 
-    def execute_avoid_obstacle(self):
-        # TODO: SET BACK TO ORIGINAL MISSION (TRAVERSE OR RETURN)
+    def track_obstacle(self):
         pass
 
+    def execute_avoid_obstacle(self, dist_to_goal, prev_phase):
+        pass
+
+    # Currently, algorithm only supports using the right side sensors of the robot for boundary following.
+    # When encountering an obstacle, robot will begin boundary following by going clockwise around the object.
+    # This was arbitrarily decided since currently there is no better way to determine which direction the robot should
+    # navigate around the obstacle.
+    def execute_boundary_following(self, min_dist):
+        front_dist = self.rf_ultrasonic.distance()
+        back_dist = self.rb_ultrasonic.distance()
+        margin = 1 
```

Comment:
how are we choosing this margin value?

Reply:
Right now we set it as 1 arbitrarily, but in the future it will be the extra padding distance in meters for the width of the robot to ensure there are no problems boundary following (i.e. the side of the robot clipping the obstacle).